awx/main/tasks/host_indirect.py

```diff
@@ -49,18 +49,19 @@ def build_indirect_host_data(job: Job, job_event_queries: dict[str, dict[str, st
         if 'res' not in event.event_data:
             continue
 
-        if 'resolved_action' not in event.event_data or event.event_data['resolved_action'] not in job_event_queries.keys():
+        if not (resolved_action := event.event_data.get('resolved_action', None)):
             continue
 
-        resolved_action = event.event_data['resolved_action']
+        resolved_action_no_module = resolved_action.rsplit('.', 1)[0]
 
-        # We expect a dict with a 'query' key for the resolved_action
-        if 'query' not in job_event_queries[resolved_action]:
+        # a.b.c --> a.b
+        # e.f.* --> e.f
+        job_event_queries_no_module = {k.rsplit('.', 1)[0]: v for k, v in job_event_queries.items()}
```

Comment:
Same as above, would the keys ever NOT be of the form `a.b.c`?

Reply:
They should not, but we can't guarantee that query files don't contain invalid entries.